mongo/single_result.go

```diff
@@ -29,20 +28,25 @@ type SingleResult struct {
 	reg *bsoncodec.Registry
 }
 
-// NewSingleResultFromBytes creates a SingleResult with an underlying Cursor pre-loaded with the provided contents.
-func NewSingleResultFromBytes(contents []byte) *SingleResult {
-	c := &Cursor{
-		bc:       driver.NewBatchCursorFromBytes(contents),
-		registry: bson.DefaultRegistry,
-		Current:  bson.Raw(contents),
+// NewMockSingleResult creates a SingleResult with the provided error, registry, and an underlying Cursor pre-loaded with
+// the provided documents, error and registry. If no registry is provided, bson.DefaultRegistry will be used.
+//
+// The documents parameter must be a slice of documents.
```

Comment:
I suggest making `documents` one `interface{}` instead of a slice since `SingleResult represents one document or zero on error.

Reply:
Sounds good. I do now require that the document is non-nil.